Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelTask.kt

```diff
@@ -169,6 +177,34 @@ class RiskLevelTask @Inject constructor(
         }
     }
 
+    private suspend fun determineRisk(
+        appConfig: ExposureWindowRiskCalculationConfig,
+        exposureWindows: List<ExposureWindow>
+    ): AggregatedRiskResult {
+        val riskResultsPerWindow =
+            exposureWindows.mapNotNull { window ->
+                riskLevels.calculateRisk(appConfig, window)?.let { window to it }
+            }.toMap()
+
+        if (analyticsSettings.analyticsEnabled.value) {
+            collectAnalyticsData(riskResultsPerWindow)
+        }
+
+        return riskLevels.aggregateResults(appConfig, riskResultsPerWindow)
+    }
+
+    private suspend fun collectAnalyticsData(riskResultsPerWindow: Map<ExposureWindow, RiskResult>) {
+        riskResultsPerWindow.keys.forEach { window ->
+            riskResultsPerWindow[window]?.let { result ->
+                val contribution = createAnalyticsExposureWindow(
+                    window,
+                    result
+                )
+                analyticsExposureWindowRepository.addNew(contribution)
+            }
+        }
+    }
+
```

Comment:
I would prefer it if we lower the analytics specific code in `RiskLevelTask`, e.g. inject a single analytics specific class on which we call a single function  e.g. `donorModule.maybeTrack(riskResultsPerWindow)` with the data it needs checking `analyticsEnabled` and using `analyticsExposureWindowRepository` would happen inside that class.

Reply:
I extracted the code to AnalyticsExposureWindowCollector